Framework/Core/src/DataOutputDirector.cxx

```diff
@@ -96,13 +97,111 @@ std::string SpectoString(InputSpec input)
 
 void DataOutputDirector::readSpecs(std::vector<InputSpec> inputs)
 {
-
   for (auto input : inputs) {
     auto s = SpectoString(input);
     readString(s);
   }
 }
 
+std::tuple<std::string, std::string, int> DataOutputDirector::readJson(std::string const& fnjson)
```

Comment:
Please split it in two and provide a string based API so that it can be tested. You should also provide a few tests with good and bad files to make sure the parsing works as expected.

Reply:
done, added also a test - test_DataOutputDirector.cxx